attn_gradcheck.py

```diff
@@ -0,0 +1,45 @@
+import torch
+
+class Attn(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, q, k, v):
+        attn = torch.matmul(q, k.swapaxes(0, 1))
```

Comment:
note that in general, we would add error checking here to make sure that we don't get un-expected size.